### PR TITLE
[WIP - do not merge] feat: Add support for Quasar components

### PR DESCRIPTION
**Pull request by rstoenescu**
Hi @octref ,

First of all, thanks for all your hard work!
Would like to add support for [Quasar Framework](http://quasar-framework.org/).

Please take a preliminary look to tell me if anything needs changing, but do not merge yet. First want to release v0.15 (which might take until end of month).

Also, are there docs for tags and attributes properties? I'd like to add more information to the Quasar helpers but don't really know what and how Vetur supports. Sorry if I missed the link on this. For example, I'd like to supply default values for certain props, offer valid options for others, and so on.

Thank you,
Razvan Stoenescu

### CODE REVIEW

**Comment by rstoenescu** [comment]
@octref also, is there (or will be) support for directives?

**Comment by octref** [comment]
@rstoenescu 

Thanks for the contribution!

As @HerringtonDarkholme said, there is no official format yet, you can take a look at https://github.com/ElementUI/element-helper-json for the format. For `-tags.json` files the most important ones are `description` & `attributes`. For `-attributes.json` important ones are `description` and `options`.

There are some discussions on standardizing a format: https://github.com/vuejs/vue/issues/7186. Would love your input over there.

> also, is there (or will be) support for directives?

Do you mean custom directives? This is certainly something we could add. Do you have any usage examples?

File: server/src/modes/template/tagProviders/index.ts [hunk 1]
```diff
@@ -54,6 +62,12 @@ export function getTagProviderSettings(workspacePath: string | null | undefined)
     if (packageJson.dependencies['vuetify']) {
       settings['vuetify'] = true;
     }
+    if (
+      packageJson.dependencies['quasar-framework'] ||
+      packageJson.dependencies['quasar-cli']
```
Comment (octref) [inline server/src/modes/template/tagProviders/index.ts]: Why are you checking `quasar-cli` here? It seems to be a global npm module. Here is's checking the dependencies in the local workspace.

Comment (rstoenescu) [reply server/src/modes/template/tagProviders/index.ts]: Things change in v0.15. The Quasar starter kit will have 'quasar-cli' as a dependency (which is not global but project folder specific). So this is correct.

Comment (HerringtonDarkholme) [reply server/src/modes/template/tagProviders/index.ts]: Does `cli` or `framework` alone can provide components? Should `||` be changed to `&&`? I wonder.

**Comment by rstoenescu** [comment]
@octref Hi, I'm a little overloaded with work at the moment. Will get back to you in a few days. Just notifying you so you won't think I do a hit and run here :)

Regarding the custom directives, it would be cool to have autocompletion for them too. Example: Quasar has touch directives ([v-touch-pan](http://quasar-framework.org/components/panning.html), [v-touch-swipe](http://quasar-framework.org/components/swiping.html), [v-touch-hold](http://quasar-framework.org/components/touch-hold.html)).

Would love to offer my input on the standard. Will fill in my comments on the vue issue as soon as I get a chance. Thanks again for this awesome tool!

**Comment by octref** [comment]
Seems 0.15 has been out for a while. Let me know when you think the helper is ready.

**Comment by octref** [comment]
@rstoenescu Can you update this PR? Seems Quasar is already on 0.16.

**Comment by rstoenescu** [comment]
@octref Will do. Give me a few days. Sorry for the delay.

**Comment by octref** [comment]
@rstoenescu https://github.com/vuejs/vetur/pull/824 was already merged — can you update the PR and get a MVP out so users could get some Quasar support, which is much better than no support? We all want something better achieved through https://github.com/vuejs/vue/issues/7186, but don't let perfection get in the way of shipping.

**Comment by rstoenescu** [comment]
New, better PR on this: https://github.com/vuejs/vetur/pull/865
Sorry for all my delays. Got a ton of work on my plate regarding Quasar.

**Pull request closed without merging**